pcdsdevices/sequencer.py

```diff
@@ -189,3 +199,131 @@ def stop(self):
+
+    def read(self):
+        """Read this line of the event sequence. 
+
+        Parameters
+        ----------
+        None.
+
+        Examples
+        --------
+        SequenceLine.read()
+
+        """
+        event_code = self.ec.get()
+        beam_delta = self.bd.get()
+        fiducial_delta = self.fd.get()
+        burst_count = self.bc.get()
+        
+        line = [event_code, beam_delta, fiducial_delta, burst_count]
+
+        return line
+
+    def write(self, line):
+        """Write to this line of the event sequence. 
+
+        Parameters
+        ----------
+        line: list
+            Four item list containing the desired values for event code,
+            beam delta, fiducial delta, and burst count, respectively, e.g.
+            [<event code>, <beam delta>, <fiducial delta>, <burst count>].
+
+        Examples
+        --------
+        SequenceLine.write([140, 12, 0, 0])
+
+        """
+
+        if len(line) != 4:
+            raise ValueError("The sequence line must be a 4 item list!")
+        else:
+            self.ec.put(line[0])        
+            self.bd.put(line[1])        
+            self.fd.put(line[2])        
+            self.bc.put(line[3])        
+
+class EventSequence(Device):
+    """Class for the event sequence of the event sequencer."""
+
+    def __init__(self, prefix, hutch_num, **kwargs):
+        
+        self._hutch_num = hutch_num
+      
+        self._lines = [] 
+        for i in range(0,20,1):
+            line_num = str(i)
+            # Pad 1 digit numbers with 0
+            if len(line_num) == 1:
+                line_num = '0' + line_num
+            line = SequenceLine(prefix, hutch_id=self._hutch_num, line=line_num, name='line{}'.format(line_num))
+            self._lines.append(line)
+        
+        super().__init__(self._hutch_num, **kwargs)
+
+    def read(self):
+        """Read the current event sequence.
+
+        Parameters
+        ----------
+        None.
+
+        Examples
+        --------
+
+        EventSequence.read()
+        
+        """
+        sequence = []
+        for line in self._lines:
+            seq_line = line.read()
+            sequence.append(seq_line)
+
+        return sequence
+
+    def write(self, sequence):
+        """Write a sequence to the event sequencer. Takes a list of lists,
+        with each sub-list representing one line of the event sequence. 
+
+        Parameters
+        ----------
+        sequence: list
+            List of lists describing the event sequence. The list cannot be
+            longer than 20 entries.
+
+        Examples
+        --------
+        seq = [[167, 19, 0, 0],
+               [168,  4, 0, 0],
+               [182,  1, 0, 0],
+               [176,  0, 0, 0],
+               [169,  0, 0, 0]]
+
+        EventSequence.write(seq)
+        
+        """
+
+        if len(sequence) > 20:
+            raise ValueError("The sequence length cannot be longer than 20!")
```

Comment:
Despite the GUI having only 20 lines, the max sequence length is actually 2048. You just need to set the `stop at step` field  (`LEN` pv) to a higher value